### PR TITLE
Add-AzureRmAccount - Add position to Credential, update parameter sets

### PR DESCRIPTION
**Pull request by cormacpayne**
## Comments

Fix for issue https://github.com/Azure/azure-powershell/issues/2048 and https://github.com/Azure/azure-powershell/issues/1102

Add position for Credential parameter so that the following is allowed:

``` powershell
Add-AzureRmAccount (Get-Credential)
```

Updated the parameter sets so that SubscriptionId and SubscriptionName are mutually exclusive. Parameter sets were split into two:
- SubscriptionId is included but not required
- SubscriptionName is include and required

This allows for either parameter to be used in a parameter set, but not both.

---

This checklist is used to make sure that common issues in a pull request are covered by the creator. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).

Below in **Overall Changes**, check off the boxes that apply to your PR. For the categories that you did not check off, you can remove them from this body. Within each of the categories that you did select, make sure that you can check off **all** of the boxes. 

For information on cleaning up the commits in your pull request, click [here](../documentation/cleaning-up-commits.md).
## Overall Changes
- [x] [**MANDATORY** - General changes](#general)
- [x] [**MANDATORY** - Add/remove/edit test(s)](#tests)
- [ ] [Add/remove/edit cmdlet(s)](#cmdlet-signature)
- [x] [Add/remove/edit parameter(s)](#parameters)
- [ ] [Edit pipeline parameters](#parameters-and-the-pipeline)
### General
- [x] Title of the PR is clear and informative
- [x] There are a small number of commits that each have an informative message
- [x] If it applies, references the bug/issue that the PR fixes
- [x] All files have the Microsoft copyright header
- [x] Cmdlets refer to management libraries through nuget references - no dlls are checked in
- [x] The PR does not introduce breaking changes (unless a major version change occurs in the assembly and module)
### Tests
- [x] PR includes test coverage for the included changes
- [x] Tests must use xunit, and should either use Moq to mock management client calls, or use the scenario test framework
- [x] PowerShell scripts used in tests must not use hard-coded values for location
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for existing resources
- [x] Tests should not use App.config files for settings
- [x] Tests should use the built-in PowerShell functions for generating random names when unique names are necessary - this will store names in the test recording
- [x] Tests should use Start-Sleep to pause rather than Thread.Sleep
### Parameters
- [x] Cmdlets should have no more than four positional parameters
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet
- [x] Parameters should be explicitly marked as Mandatory or not, and should contain a HelpMessage
- [x] No parameter is of type `object`.
  - Management cmdlets should have the following parameters and aliases:
    - ResourceGroupName with (optional) alias to `ResourceGroup` type string marked as [ValueFromPipelineByPropertyName]
    - Name with alias to `ResourceName` type string marked as [ValueFromPipelineByPropertyName]
    - Location (if appropriate) type string
    - Tag, type `HashTable`

### CODE REVIEW

File: src/ResourceManager/Profile/Commands.Profile/Account/AddAzureRmAccount.cs [hunk 1]
```diff
@@ -34,12 +34,14 @@ namespace Microsoft.Azure.Commands.Profile
     [OutputType(typeof(PSAzureProfile))]
     public class AddAzureRMAccountCommand : AzureRMCmdlet, IModuleAssemblyInitializer
     {
-        private const string UserParameterSet = "User";
-        private const string ServicePrincipalParameterSet = "ServicePrincipal";
-        private const string ServicePrincipalCertificateParameterSet = "ServicePrincipalCertificate";
-        private const string AccessTokenParameterSet = "AccessToken";
-        private const string SubscriptionNameParameterSet = "SubscriptionName";
-        private const string SubscriptionIdParameterSet = "SubscriptionId";
+        private const string UserParameterSetWithSubscriptionId = "UserWithSubscriptionId";
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Account/AddAzureRmAccount.cs]: Need to update the DefaultParameterSetName in the Cmdlet attribute to match one of the actual parameter sets (UserWithSubscriptionId is the best bet)

File: src/ResourceManager/Profile/Commands.Profile/Account/AddAzureRmAccount.cs [hunk 2]
```diff
@@ -50,56 +52,60 @@ public class AddAzureRMAccountCommand : AzureRMCmdlet, IModuleAssemblyInitialize
 
         public string EnvironmentName { get; set; }
 
-
-        [Parameter(ParameterSetName = UserParameterSet, Mandatory = false, HelpMessage = "Optional credential")]
-        [Parameter(ParameterSetName = ServicePrincipalParameterSet, Mandatory = true, HelpMessage = "Credential")]
-        [Parameter(ParameterSetName = SubscriptionIdParameterSet, Mandatory = false, HelpMessage = "Optional credential")]
-        [Parameter(ParameterSetName = SubscriptionNameParameterSet, Mandatory = false, HelpMessage = "Optional credential")]
+        
+        [Parameter(ParameterSetName = UserParameterSetWithSubscriptionId, Mandatory = false, HelpMessage = "Optional credential", Position = 0)]
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Account/AddAzureRmAccount.cs]: Note: if there wasn't another change to make I would not make this comment.  Please break the Parameter attributes over multiple lines so they are not so difficult to read in the default code review pane.

**Comment by cormacpayne** [comment]
@markcowl addressed code review comments

**Comment by zephrxx** [comment]
Doesn't work for 2 factor auth

**Pull request merged**